### ISSUE TITLE
Issue with spring boot 3 entities and kotlin

### ISSUE THREAD

**Issue by rburgst** [spring-io/initializr#1359]
I created an empty project with gradle, h2, spring-data-jpa, kotlin, spring boot3, graphql and webmvc.

The resulting app seems to have problems with the hibernate l1 cache (more specifically I have a 1:n relationship between Club and Athlete, when the list of clubs is already loaded and I load a list of athletes, their referenced clubs will still be hibernate proxies even though all clubs are already loaded).
I tinkered around with the build script and added a bunch of kotlin compiler plugins etc but so far the app still shows problems with hibernate l1 cache.

I created the very same application in java with the same settings and there, everything works perfectly. So I am not 100% what the reason for these issues is
a) the build script is still missing something
b) the implementation of some of the entity methods is off (or their annotations)
c) some other problem

My problem is that I dont even know in which project I should create the issue (kotlin, spring, hibernate).

Here you can find a sample repo: 
https://github.com/rburgst/spring-boot3-graphql-kotlin

**Comment by rburgst** [comment]
ok, I finally figured out what the problem is, I have set 
```
spring.jpa.open-in-view=false
```
which causes  the list of clubs passed in as a parameter to be from a different hibernate session than the athletes loaded in the current session.
```kotlin
    @BatchMapping(typeName = "Club")
    fun athletes(clubs: List<Club>): Map<Club, List<Athlete>> {
        val clubIdList = clubs.map { it.id!! }.toList()
        val athletes = athleteRepository.findAthletesByClubIdIn(clubIdList)
        val result = athletes
            .groupBy { it.club!! }
        return result
    }
```
Therefore, the `athlete.club` will return a hibernate proxy which is not initialized. 

Either you make the `@BatchMapping` method `@Transactional` which causes the `athlete.club` instances to be hydrated again (therefore, defeating the purpose of the `@BatchMapping` as this is effectively an n+1 on the `club` table) or you simply get an exception that the hashcode of the `athlete.club` cannot be calculated.

I feel that the API for `@BatchMapping` is a bit unfortunate, it would have been better to allow 
```kotlin
    fun athletes(clubs: List<Club>): Map<Tsid, List<Athlete>>
```
